Sources/XcodesKit/XcodeInstaller.swift

```diff
@@ -132,6 +135,47 @@ public final class XcodeInstaller {
     }
 
     public func install(_ installationType: InstallationType) -> Promise<Void> {
+        return firstly { () -> Promise<InstalledXcode> in
+            return self.install(installationType, attemptNumber: 0)
+        }
+        .done { xcode in
+            Current.logging.log("\nXcode \(xcode.version.descriptionWithoutBuildMetadata) has been installed to \(xcode.path.string)")
+            Current.shell.exit(0)
+        }
+    }
+    
+    private func install(_ installationType: InstallationType, attemptNumber: Int) -> Promise<InstalledXcode> {
+        return firstly { () -> Promise<(Xcode, URL)> in
+            return self.getXcodeArchive(installationType)
+        }
+        .then { xcode, url -> Promise<InstalledXcode> in
+            return self.installArchivedXcode(xcode, at: url)
+        }
+        .recover { error -> Promise<InstalledXcode> in
+            switch error {
+            case XcodeInstaller.Error.damagedXIP(let damagedXIPURL):
+                guard attemptNumber <= 1 else { throw error }
```

Comment:
Won't this retry twice? When `attemptNumber` is 0 and 1?

Reply:
😅 sure does, thank you :) changed to `< 1`